extensions/sql-migration/src/main.ts

```diff
@@ -5,9 +5,22 @@
 
 import * as vscode from 'vscode';
 import { DashboardWidget } from './dashboard/sqlServerDashboard';
+import * as constants from './constants/strings';
+import { ServiceClient } from './service/serviceClient';
+import { migrationServiceProvider } from './service/provider';
 
 let widget: DashboardWidget;
 export async function activate(context: vscode.ExtensionContext): Promise<DashboardWidget> {
+	if (!migrationServiceProvider) {
+		vscode.window.showErrorMessage(constants.serviceProviderInitializationError).then(() => { }, () => { });
+	}
+	// asynchronously starting the service
+	const outputChannel = vscode.window.createOutputChannel(constants.serviceName);
+	const serviceClient = new ServiceClient(outputChannel);
+	serviceClient.startService(context).catch((e) => {
```

Comment:
Why are you not awaiting here? If the service client fails to start the extension is pretty much useless so might as well fail activation.

Reply:
That will prevent the initialization of the dashboard. That's why I have not blocking on it.

Reply:
I would still suggest saving all the "startup" promises in vars and then awaiting Promise.all so that the extension will fail activation if a critical error happens